pac4j-http/src/main/java/org/pac4j/http/credentials/authenticator/DigestAuthenticator.java

```diff
@@ -0,0 +1,15 @@
+package org.pac4j.http.credentials.authenticator;
+
+import org.pac4j.http.credentials.DigestCredentials;
+import org.pac4j.http.credentials.UsernamePasswordCredentials;
+
+/**
+ * This interface represents the contract to validate digest credentials.
+ *
+ * @author Mircea Carasel
+ */
+public interface DigestAuthenticator extends Authenticator<DigestCredentials> {
```

Comment:
That is an implementation of the DigestAuthenticator interface that should be done in the user application. Example:
public class SecurityDigestAuthenticator implements DigestAuthenticator {

    @Override
    public void validate(DigestCredentials credentials) {
        if (credentials == null) {
            this.throwsException("No credential");
        }
        String username = credentials.getUsername();
        if (CommonHelper.isBlank(username)) {
            this.throwsException("Username cannot be blank");
        }

        HttpProfile profile = new HttpProfile();
        profile.setId(username);
        credentials.setUserProfile(profile);
    }
    protected void throwsException(String message) {
        throw new CredentialsException(message);
    }

I followed the same model as for the Basic direct authenticator.  The purpose is to inject the http user profile here